projects/cds/src/profiletag/services/profile-tag-push-events.service.ts

```diff
@@ -0,0 +1,372 @@
+ * The service observes the event service and the active cart service for supported events. These events are parsed into
+ * a profiletag compliant format and enriched by segments and actions from the latest personalization context.
+ *
+ * Currently supported events from the event service:
+ *  - CartPageVisited
+ *  - CategoryPageVisited
+ *  - HomePageVisited
+ *  - KeywordSearchPageVisited
+ *  - OrderConfirmationPageVisited
+ *  - PageVisited
+ *  - ProductDetailsPageVisited
+ *  - CartAddEntrySuccessEvent
+ *  - CartRemoveEntrySuccessEvent
+ *  - CartUpdateEntrySuccessEvent
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class ProfileTagPushEventsService {
+  private pushEvents$: Observable<ProfileTagPushEvent> = merge(
+    this.categoryPageVisited(),
+    this.productDetailsPageView(),
+    this.searchResultsChanged(),
+    this.homePageVisitedEvent(),
+    this.cartPageVisitedEvent(),
+    this.navigatedEvent(),
+    this.orderConfirmationPageVisited(),
+    this.addedToCart(),
+    this.removedFromCart(),
+    this.modifiedCart()
+  );
+
+  constructor(
+    protected eventService: EventService,
+    protected personalizationContextService: PersonalizationContextService
+  ) {}
+
+  /**
+   * Returns a push event emitting observable that emits all converted events emitted by the event or the active cart service.
+   * These events are enriched with segments and actions from the latest personalization context.
+   *
+   * @returns an observable emitting profiletag push events
+   */
+  getPushEvents(): Observable<ProfileTagPushEvent> {
+    return this.pushEvents$.pipe(
+      withLatestFrom(
+        merge(
+          of({ segments: undefined, actions: undefined }),
+          this.personalizationContextService.getPersonalizationContext()
+        )
+      ),
+      map(([item, personalizationContext]) => {
+        item.data = item.data ? item.data : {};
+        item.data.segments = personalizationContext.segments;
+        item.data.actions = personalizationContext.actions;
+        return item;
+      })
+    );
+  }
+
+  /**
+   * Adds a new push event emitting observable to this service. This observable will be merged with the internal one.
+   * This method can be used to extend the functionality of this service at runtime.
+   *
+   * @param event an observable emitting profiltag push events
+   */
+  addPushEvent(event: Observable<ProfileTagPushEvent>): void {
+    this.pushEvents$ = merge(this.pushEvents$, event);
+  }
+
+  /**
+   * Emits the category page visited event.
+   *
+   * @returns an observable emitting events that describe category page visits in a profiltag compliant way
+   * @see CategoryPageResultsEvent
+   * @see CategoryViewPushEvent
+   */
+
+  protected categoryPageVisited(): Observable<ProfileTagPushEvent> {
+    return this.eventService.get(CategoryPageResultsEvent).pipe(
+      tap((_) => {
+        console.log(_);
```

Comment:
Let's remove this

Reply:
Thats embarrassing! I usually remember to remove these....